### PR TITLE
CASSANDRA-21548: cqlsh prompt does not reset to default after dropping the current keyspace

### PR DESCRIPTION
**Pull request by arvindKandpal-ksolves**
### Summary
This PR fixes an issue where the `cqlsh` shell prompt remains stuck displaying a keyspace name even after that keyspace has been successfully dropped.

### Problem
When a user is currently inside a keyspace in `cqlsh` (e.g., `cqlsh:test_ks>`) and executes `DROP KEYSPACE test_ks;`, the server deletes the keyspace, but the client-side `cqlsh` prompt does not reset to the default `cqlsh>` prompt. Executing subsequent queries inside the dropped keyspace context throws a `Keyspace does not exist` error.

### Solution
- Updated `perform_simple_statement` in `pylib/cqlshlib/cqlshmain.py` to check if an executed statement is a `DROP KEYSPACE` statement.
- If the dropped keyspace matches `self.current_keyspace`, `self.current_keyspace` is reset to `None`, which automatically updates the shell prompt back to `cqlsh>`.
- Included safe string manipulation and `.lower()` normalization to handle edge cases like quoted keyspace names (`"Test_KS"`) and `IF EXISTS` clauses.

### Testing
- Added automated unit tests in `pylib/cqlshlib/test/test_cqlsh_output.py` covering:
  1. Standard keyspace drop resetting the prompt.
  2. Dropping a non-active keyspace without altering the current prompt.
  3. Case-sensitive quoted keyspace names and `DROP KEYSPACE IF EXISTS` syntax.
- Verified test execution using pytest (`python3 -m pytest pylib/cqlshlib/test/test_cqlsh_output.py -k "test_drop_keyspace_prompt_behavior"`).

---

patch by Arvind Kandpal; reviewed by TBA for CASSANDRA-21548

### CODE REVIEW

File: pylib/cqlshlib/cqlshmain.py [hunk 1]
```diff
@@ -911,6 +911,25 @@ def perform_simple_statement(self, statement):
...
+            # stay case-sensitive, unquoted ones fold to lower case).
+            qs = getattr(statement, 'query_string', str(statement))
+            if qs.lstrip().lower().startswith('drop'):
+                try:
+                    parsed = ruleset.cql_parse(qs)[1]
+                except IndexError:
+                    parsed = None
+                if (parsed is not None and len(parsed.matched) >= 2
+                        and parsed.matched[0][1].upper() == 'DROP'
+                        and parsed.matched[1][1].upper() == 'KEYSPACE'):
```
Comment (driftx) [inline pylib/cqlshlib/cqlshmain.py:929]: what about SCHEMA?

Comment (arvindKandpal-ksolves) [reply pylib/cqlshlib/cqlshmain.py:929]: Hi @driftx,

I tried testing DROP SCHEMA locally on trunk, but it throws a SyntaxException:

```Plaintext
cqlsh:test_schema> DROP SCHEMA test_schema;
SyntaxException: line 1:5 no viable alternative at input 'SCHEMA' ([DROP] SCHEMA...)
```

Since I couldn't find the server-side logic for DROP SCHEMA in the latest trunk, I just wanted to ask that should we add SCHEMA here primarily for backward compatibility with older clusters?

Also, because of this SyntaxException on trunk, we cannot directly add a test case for DROP SCHEMA in the current test suite, as it will naturally fail. What we do here ? 

Please let me know !

Comment (driftx) [reply pylib/cqlshlib/cqlshmain.py:929]: Ah sorry, I didn't check trunk and it doesn't look like it needs it.

> because of this SyntaxException on trunk, we cannot directly add a test case for DROP SCHEMA in the current test suite, as it will naturally fail. What we do here ?

If we are only fixing this on trunk then we don't need to be concerned.  If we were fixing it on branches where DROP SCHEMA is present, then we'd handle it via multiple PRs.

Comment (smiklosovic) [reply pylib/cqlshlib/cqlshmain.py:929]: @arvindKandpal-ksolves I think just trunk is fine (6.0 + trunk)
